code/game/machinery/computer/atmos_alert.dm

```diff
@@ -7,33 +7,44 @@
 	light_color = LIGHT_COLOR_CYAN
 	// List of alarms and their state in areas. This is sent to TGUI
 	var/list/alarm_cache
+	var/parent_area_type = null
 
 /obj/machinery/computer/atmos_alert/Initialize(mapload)
 	. = ..()
 	alarm_cache = list()
 	alarm_cache["minor"] = list()
 	alarm_cache["priority"] = list()
 
+	parent_area_type = (get_area(src)).type
+
+	if(parent_area_type in subtypesof(/area/ruin))
+		// figure out which ruin we are on
+		while(!(type2parent(parent_area_type) in GLOB.ruin_prototypes))
+			parent_area_type = type2parent(parent_area_type)
+
+	else if(parent_area_type in subtypesof(/area/station))
+		parent_area_type = /area/station
+	else
+		parent_area_type = null
+
+
 /obj/machinery/computer/atmos_alert/process()
 	// This is relatively cheap because the areas list is pretty small
-	for(var/area/A as anything in GLOB.all_areas)
-		if(!A.master_air_alarm)
-			continue // No alarm
-		if(A.master_air_alarm.z != z)
-			continue // Not on our z-level
-		if(!A.master_air_alarm.report_danger_level)
+	for(var/obj/machinery/alarm/AirAlarm as anything in GLOB.air_alarms)
```

Comment:
Camel-case this var, please.

Reply:
you mean airAlarm?

Reply:
Style guidelines say all snake_case no exceptions, and Burza confirmed as much so I'm converting to that.